src/Logic/Article/ArticleManager.php

```diff
@@ -0,0 +1,115 @@
+     * @param \Repository\Article\Postgresql\PostgresqlCollection $articleCollectionRepository
+     * @param \Repository\ArticleToCategory\Postgresql\PostgresqlCollection $articleToCategoryCollectionRepository
+     */
+    public function __construct(
+        \Logic\Article\Crud\CrudManager $articleCrudManager,
+        \Repository\Article\Postgresql\PostgresqlEntity $articleEntityRepository,
+        \Repository\Article\Postgresql\PostgresqlCollection $articleCollectionRepository,
+        \Repository\ArticleToCategory\Postgresql\PostgresqlCollection $articleToCategoryCollectionRepository
+    ) {
+        $this->articleCrudManager = $articleCrudManager;
+        $this->articleCollectionRepository = $articleCollectionRepository;
+        $this->articleEntityRepository = $articleEntityRepository;
+        $this->articleToCategoryCollectionRepository = $articleToCategoryCollectionRepository;
+    }
+
+    /**
+     * @param int $limit
+     * @param int $offset
+     * @param array $sorting
+     * @param array $filters
+     *
+     * @return \Data\Article\ArticleCollection
+     */
+    public function list($limit = 100, $offset = 0, $sorting = [], $filters = []): \Data\Article\ArticleCollection {
+        $parameters = [
+            '__subset' => [
+                'limit' => $limit,
+                'offset' => $offset,
+            ]
+        ];
+
+        if (!empty($sorting)) {
+            $parameters['__sorting'] = $sorting;
+        }
+
+        if (!empty($filters)) {
+            foreach ($filters as $entityProperty => $entityValue) {
+                $parameters[$entityProperty] = [$entityValue];
+            }
+        }
+
+        $articleCollection = \Maleficarum\Ioc\Container::get(\Data\Article\ArticleCollection::class);
+        $this->articleCollectionRepository->populate($articleCollection, $parameters);
+
+        return $articleCollection;
+    }
+
+    /**
+     * Assign article to category
+     *
+     * @param int $articleId
+     * @param array $categoryIds
+     *
+     * @throws \Maleficarum\Storage\Exception\Repository\EntityNotFoundException
+     *
+     * @return \Data\ArticleToCategory\ArticleToCategoryCollection
+     */
+    public function assignToCategories(int $articleId, array $categoryIds): \Data\ArticleToCategory\ArticleToCategoryCollection {
+        $article = $this->articleCrudManager->read($articleId);
+        $articleToCategoryCollection = \Maleficarum\Ioc\Container::get(\Data\ArticleToCategory\ArticleToCategoryCollection::class);
+        $parameters = [
+            'mapArticleCategoryArticleId' => [$articleId]
+        ];
+
+        $this->articleToCategoryCollectionRepository->populate($articleToCategoryCollection, $parameters);
+
+        try {
+            $shard = $this->articleToCategoryCollectionRepository->getShard($articleToCategoryCollection);
+            $shard->inTransaction() or $shard->beginTransaction();
```

Comment:
I would avoid opening transaction based on specific repository as it may span across multiple collections